### PR TITLE
accounts db/refactor accounts db test - add macro for accounts-db panic test

### PR DESCRIPTION
**Pull request by HaoranYi**
#### Problem

add macro for accounts-db panic test 

Continued on #784

#### Summary of Changes

1. add macro for accounts-db panic test 
2. convert double remove test for both account file provider

Fixes #

### CODE REVIEW

**Comment by HaoranYi** [comment]
The actual change is in the last commit.
Please only review the last change if you have already reviewed https://github.com/anza-xyz/agave/pull/784

File: accounts-db/src/accounts_db.rs [hunk 1]
```diff
@@ -9686,6 +9686,23 @@ pub mod tests {
         };
     }
 
+    macro_rules! define_accounts_db_test_should_panic {
```
Comment (brooksprumo) [inline accounts-db/src/accounts_db.rs]: This impl works. We could also refactor the other macro to dedup the code within the macros. Doesn't seem to be a big issue IMO.

**Review by brooksprumo — COMMENTED** [review]
Looks good. Please re-request a review once the previous PR has merged and this PR has been rebased.

File: accounts-db/src/accounts_db.rs (hunk 1 above)
Comment (HaoranYi) [reply accounts-db/src/accounts_db.rs]: Yeah. Refactored.

File: accounts-db/src/accounts_db.rs [hunk 2]
```diff
@@ -9670,35 +9670,36 @@ pub mod tests {
...
+    ///   For regular test, use the following syntax.
+    ///     define_accounts_db_test!(TEST_NAME, |accounts_db| { TEST_BODY }); // regular test
+    ///   For test that should panic, use the following syntax.
+    ///     define_accounts_db_test!(TEST_NAME, panic = "PANIC_MSG", |accounts_db| { TEST_BODY });
     macro_rules! define_accounts_db_test {
-        ($name:ident, |$accounts_db:ident| $inner: tt) => {
-            #[test_case(AccountsFileProvider::AppendVec; "append_vec")]
-            #[test_case(AccountsFileProvider::HotStorage; "hot_storage")]
-            fn $name(accounts_file_provider: AccountsFileProvider) {
+        (@testfn $name:ident, $accounts_file_provider: ident, |$accounts_db:ident| $inner: tt) => {
```
Comment (brooksprumo) [inline accounts-db/src/accounts_db.rs:9678]: Nice 👍

**Comment by HaoranYi** [comment]
No code change. Rebase to master to pick up #784.

**Review by brooksprumo — APPROVED** [review]
:shipit:

**Comment by HaoranYi** [comment]
@jeffwashington Can you help to review this PR? Thanks!

**Review by jeffwashington — APPROVED** [review]
lgtm
How many tests panic?

**Comment by HaoranYi** [comment]
> lgtm How many tests panic?

There are 9 more tests that expect panics.

**Pull request merged**